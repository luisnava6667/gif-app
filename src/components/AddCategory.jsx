import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')
  const onInputChange = (e) => {
    setInputValue(e.target.value)
  }
  const onSubmit = (e) => {
    e.preventDefault()
    const newInputValue = inputValue.trim()
    if (inputValue.trim().length <= 1) return
    // setCategories((categories) => [inputValue, ...categories])
    setInputValue('')
    onNewCategory(inputValue.trim())
  }
  return (
    <form aria-label='form' onSubmit={onSubmit}>
      <input
        type='text'
        placeholder='Buscar Gifs'
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
