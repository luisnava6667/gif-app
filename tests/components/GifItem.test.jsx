import { render, screen } from '@testing-library/react'
import { GifGridItem } from '../../src/components/GifGridItem'
describe('Pruebas en <GifGridItem/> ', () => {
  const title = 'ReactTest'
  const url = 'https://react.com/'
  test('debe de hacer match con el Snapshot', () => {
    const { container } = render(<GifGridItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })
  test('debe de mostrar la imagen con el url y el ALT indicado', () => {
    render(<GifGridItem title={title} url={url} />)
    // screen.debug()
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifGridItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
