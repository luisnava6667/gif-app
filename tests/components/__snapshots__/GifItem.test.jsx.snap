// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/>  debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="ReactTest"
      src="https://react.com/"
    />
    <p>
      ReactTest
    </p>
  </div>
</div>
`;
